modules/siddhi-extensions/string/src/test/java/org/wso2/siddhi/extension/string/EqualsIgnoreCaseFunctionExtensionTestCase.java

```diff
@@ -0,0 +1,72 @@
+package org.wso2.siddhi.extension.string;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import junit.framework.Assert;
+
+import org.apache.log4j.Logger;
+import org.junit.Before;
+import org.junit.Test;
+import org.wso2.siddhi.core.ExecutionPlanRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.query.output.callback.QueryCallback;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.util.EventPrinter;
+import org.wso2.siddhi.extension.string.test.util.SiddhiTestHelper;
+
+public class EqualsIgnoreCaseFunctionExtensionTestCase {
+    static final Logger log = Logger.getLogger(EqualsIgnoreCaseFunctionExtensionTestCase.class);
+    private AtomicInteger count = new AtomicInteger(0);
+    private volatile boolean eventArrived;
+
+    @Before
+    public void init() {
+        count.set(0);
+        eventArrived = false;
+    }
+
+    @Test
```

Comment:
Shall we add few more test cases to test validation. Ex: sending null attributes, Not sending parameters to str:equalsIgnoreCase